charts/redpanda/service.loadbalancer.go

```diff
@@ -0,0 +1,175 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=service.loadbalancer.go.tpl
+package redpanda
+
+import (
+	"fmt"
+
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+)
+
+func LoadBalancerServices(dot *helmette.Dot) []*corev1.Service {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	// This is technically a divergence from previous behavior but this matches
+	// the NodePort's check and is more reasonable.
+	if !values.External.Enabled || !values.External.Service.Enabled {
+		return nil
+	}
+
+	if values.External.Type != corev1.ServiceTypeLoadBalancer {
+		return nil
+	}
+
+	externalDNS := ptr.Deref(values.External.ExternalDNS, Enableable{})
+
+	labels := FullLabels(dot)
+	// This typo is intentionally being preserved for backwards compat
+	// https://github.com/redpanda-data/helm-charts/blob/2baa77b99a71a993e639a7138deaf4543727c8a1/charts/redpanda/templates/service.loadbalancer.yaml#L33
+	labels["repdanda.com/type"] = "loadbalancer"
+
+	selector := StatefulSetPodLabelsSelector(dot)
+
+	services := []*corev1.Service{}
+	//   {{- range $replicaIndex := untilStep 0 ($values.statefulset.replicas|int) 1 }}
+	replicas := values.Statefulset.Replicas // TODO fix me once the transpiler is fixed.
+	for i := 0; i < replicas; i++ {
+		//     {{- $podName := printf "%s-%d" (include "redpanda.fullname" $root) $replicaIndex }}
+		podname := fmt.Sprintf("%s-%d", Fullname(dot), i)
+
+		// NB: A range loop is used here as its the most terse way to handle
+		// nil maps in gotohelm.
+		annotations := map[string]string{}
+		for k, v := range values.External.Annotations {
+			annotations[k] = v
+		}
+
+		if externalDNS.Enabled {
+			prefix := podname
+			if len(values.External.Addresses) > i {
+				prefix = values.External.Addresses[i]
+			}
+
+			address := fmt.Sprintf("%s.%s", prefix, helmette.Tpl(*values.External.Domain, dot))
```

Comment:
The `helmette.Tpl` is not implemented, so that go implementation is incomplete.

Reply:
Added an implementation of `Tpl`!